### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by SuperQ**
* Enable dependabot.
* Update Go to 1.20.
* Update bingo.
* Update golangci-lint.
* Remove misspell (handled by golangci-lint).

### CODE REVIEW

**Review by saswatamcode — APPROVED** [review]
Thanks a lot! Generally LGTM! 

I think the "check 404 link" test fails now, due to the new go version printing DNS resolver address with the error. Maybe we can convert https://github.com/bwplotka/mdox/blob/main/pkg/mdformatter/linktransformer/link_test.go#L311 to use `testutil.Assert` with `strings.Contains` instead, as that might be simplest way to fix? 🙂

**Comment by SuperQ** [comment]
@saswatamcode Done, updated test.

File: pkg/mdformatter/linktransformer/link_test.go [hunk 1]
```diff
@@ -308,9 +304,7 @@ func TestValidator_TransformDestination(t *testing.T) {
 			MustNewValidator(logger, []byte(""), anchorDir, nil),
 		))
 		testutil.NotOk(t, err)
-		testutil.Equals(t, fmt.Sprintf("%v: 2 errors: "+
-			"%v:1: \"https://docs.gfoogle.com/drawings/d/e/2PACX-1vTBFK_cGMbxFpYcv/pub?w=960&h=720\" not accessible even after retry; status code 0: Get \"https://docs.gfoogle.com/drawings/d/e/2PACX-1vTBFK_cGMbxFpYcv/pub?w=960&h=720\": dial tcp: lookup docs.gfoogle.com: no such host; "+
-			"%v:1: \"https://bwplotka.dev/does-not-exists\" not accessible; status code 404: Not Found", tmpDir+filePath, relDirPath+filePath, relDirPath+filePath), err.Error())
+		testutil.Assert(t, strings.Contains(err.Error(), "status code 404: Not Found"))
```
Comment (saswatamcode) [inline pkg/mdformatter/linktransformer/link_test.go:307]: Thanks! This test is meant to check if we are including filename, line number and the faulty link in the error, so maybe we can do something like this? 🙂  
```suggestion
		testutil.Assert(t, strings.Contains(err.Error(), fmt.Sprintf("%v:1: \"https://docs.gfoogle.com/drawings/d/e/2PACX-1vTBFK_cGMbxFpYcv/pub?w=960&h=720\" not accessible even after retry; status code 0",relDirPath+filePath)))
		testutil.Assert(t, strings.Contains(err.Error(), fmt.Sprintf("%v:1: \"https://bwplotka.dev/does-not-exists\" not accessible; status code 404: Not Found", relDirPath+filePath)))
		
```

Comment (saswatamcode) [reply pkg/mdformatter/linktransformer/link_test.go:307]: I'll commit this and merge!

**Comment by saswatamcode** [comment]
Ah missed the path!

File: pkg/mdformatter/linktransformer/link_test.go (hunk 1 above)
Comment (SuperQ) [reply pkg/mdformatter/linktransformer/link_test.go]: Ahh, thanks, I thought the test was just for the 404 as the name suggested.

**Comment by SuperQ** [comment]
One thing that can also be done now is switch to the Go 1.20 multi-error handling.

**Pull request merged**